Mentorship_Problems/Problem-1.js

```diff
@@ -0,0 +1,246 @@
+  arr.map((item) => {
+    console.log(item);
+  });
+};
+printAllDriversInformation(listOfDrivers);
+
+// 2. Given a driver id, write a function to return the matched driver.
+
+const printDriverWithId = (arr, id) => {
+  const driver = arr.filter((item) => item.id === id);
+  console.log(driver);
+};
+printDriverWithId(listOfDrivers, 1010);
+
+// 3. Write a function to find the driver with the longest trip.
+
+const findDriverWithLongestTrip = (arr) => {
+  let maxDuration = arr[0].duration;
+  for (let i = 0; i < arr.length; i++) {
+    if (maxDuration < arr[i].duration) {
+      maxDuration = arr[i].duration;
+    }
+  }
+  let driver = arr.filter((driver) => driver.duration === maxDuration);
+  console.log(`${driver[0].first_name} ${driver[0].last_name}`);
+};
+
+findDriverWithLongestTrip(listOfDrivers);
+
+// 4. Write a function to find the driver with the shortest trip.
+
+const findDriverWithShortestTrip = (arr) => {
+  let minDuration = arr[0].duration;
+  for (let i = 0; i < arr.length; i++) {
+    if (minDuration > arr[i].duration) {
+      minDuration = arr[i].duration;
+    }
+  }
+  let drivers = arr.filter((driver) => driver.duration === minDuration);
+  let driverList = drivers.map(
+    (driver) => `${driver.first_name} ${driver.last_name}`
+  );
+  console.log(driverList);
+};
+
+findDriverWithShortestTrip(listOfDrivers);
+
+// 5. Write a function to sort the list of drivers from longest to shortest trip.
+
+const sortListOfDriversFromLongestToShortest = (arr) => {
+  let driversList = arr.map((driver) => driver.duration);
+  let sortedDurationList = driversList.sort((a, b) => b - a);
+  console.log(sortedDurationList);
+};
+
+console.log(sortListOfDriversFromLongestToShortest(listOfDrivers));
+
+sortListOfDriversFromLongestToShortest(listOfDrivers);
+
+// 6. A wife called to look for husband and claiming the driver works here. Please write a function
+//       to find the driver when a first name is given as parameters.
+
+const findDriverWithFirstName = (arr, firstName) => {
+  const driver = arr.filter(
+    (driver) => driver.first_name.toLowerCase() === firstName.toLowerCase()
+  );
+  console.log(driver);
+};
+
+findDriverWithFirstName(listOfDrivers, 'thor');
+
+// 7. More trouble coming in, police is now looking for a convict. They can provide any form of a full name format such as
+//       "John Hansom" or "Hansom, John". Write a function to peform a search on the list of drivers
+//       and return the driver if found matching. Return null if no one found.
+
+const searchDriverWithFullName = (arr, name) => {
+  const driverName = arr.filter((driver) => {
+    const fullName = driver.first_name + ' ' + driver.last_name;
+    const fullName1 = driver.last_name + ', ' + driver.first_name;
+
+    if (name === fullName) {
+      return driver;
+    } else if (name === fullName1) {
+      return driver;
+    } else {
+      return null;
+    }
+  });
+  console.log(driverName);
```

Comment:
The solution works however the more appropriate implementation would be 
1. breaking down the input name into `first name` and `last name`
2. iterate over the drivers array and compare both first name and last name with `&&` operator.